### PR TITLE
#2415 - Use box_approximation instead of overapproximate(X, Hyperrectangle)

### LINKED ISSUE #2415: Use box_approximation instead of overapproximate(X, Hyperrectangle)
In #1556 we unified `box_approximation` and `overapproximate(X, Hyperrectangle)`:

```julia
box_approximation(X) = overapproximate(X, Hyperrectangle)
```

I suggest that we actually swap them:

```julia
overapproximate(X, Hyperrectangle) = box_approximation(X)
```

The motivation is to bring some order into the `overapproximate.jl` file where we have too many methods and it is very hard to find out which special cases of the box approximation we have implemented. I find it easier to go to `box_approximation.jl` and search for the signature `box_approximation(X::MySetType)`.

### PR DESCRIPTION
**Pull request by schillic**
Closes #2415.

### CODE REVIEW

File: src/Approximations/ballinf_approximation.jl [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+### Input
+
+- `S` -- convex set
+
+### Output
+
+A tight ball in the infinity norm.
+"""
+function ballinf_approximation(S::LazySet)
+    return overapproximate(S, BallInf)
```
Comment (mforets) [inline src/Approximations/ballinf_approximation.jl:19]: i didn't expect the function in the new file to call `overapproximate(S, BallInf)`, i thought you would outsource the code here.

**Review by mforets — APPROVED** [review]
lgtm

File: src/Approximations/ballinf_approximation.jl (hunk 1 above)
Comment (schillic) [reply src/Approximations/ballinf_approximation.jl:19]: Good catch, I just copy-pasted that part. I'll change it as well!

**Comment by schillic** [comment]
```julia
  Expression: overapproximate(e, EmptySet) == overapproximate(e, 0.001) == overapproximate(e, Hyperrectangle) == overapproximate(e) == e
  MethodError: LazySets.Approximations.overapproximate(::EmptySet{Rational{Int64}}, ::Type{Hyperrectangle}) is ambiguous. Candidates:
    overapproximate(S::LazySet, ::Type{#s45} where #s45<:Hyperrectangle) in LazySets.Approximations at /home/travis/build/JuliaReach/LazySets.jl/src/Approximations/overapproximate.jl:39
    overapproximate(∅::EmptySet, options...) in LazySets.Approximations at /home/travis/build/JuliaReach/LazySets.jl/src/Approximations/overapproximate.jl:121
  Possible fix, define
    overapproximate(::EmptySet, ::Type{#s45} where #s45<:Hyperrectangle)
```
While I understand this error, I do not understand why it only occurred after the latest commit. That should not have anything to do with it. Anyway, I guess we need a fix as in #2423 for all set types...

**Review by mforets — APPROVED** [review]

**Pull request merged**